proto/cosmos/base/store/v1beta1/listening.proto

```diff
@@ -14,3 +16,19 @@ message StoreKVPair {
   bytes key        = 3;
   bytes value      = 4;
 }
+
+// BlockMetadata contains all the abci event data of a block
+// the file streamer dump them into files together with the state changes.
+message BlockMetadata {
+    // DeliverTx encapulate deliver tx request and response.
+    message DeliverTx {
+        tendermint.abci.RequestDeliverTx request = 1;
+        tendermint.abci.ResponseDeliverTx response = 2;
+    }
```

Comment:
I don't know if there's an existing type that encapsulate the deliverTx request and response.

Reply:
Yes, it's for single metadata write.
Currently the file is structured as:
```
metadata
kvpair
kvpair
...
```
I plan to change it into two files:
- block-meta
```
request_begin_block
response_begin_block
request_end_block
response_end_block
response_commit
(request_deliver_tx, response_deliver_tx)
```
- block-data
```
kv-pair
...
```

Reply:
Yeah I like the idea of two separate files actually. For the txs, it can be delineated by newlines?

Reply:
> Yeah I like the idea of two separate files actually. For the txs, it can be delineated by newlines?

since we observe the writes in commit event, we can't distinguish them by txs.
do you mean the deliver txs in block metadata?

Reply:
Yes. I mean in the file streaming implementation. Have one file for the begin/end block structure and one for txs.

Reply:
> Yes. I mean in the file streaming implementation. Have one file for the begin/end block structure and one for txs.

Do you mean three files:
- block meta, begin/end/commit block messages
- block txs, deliver tx messages
- block data, the `StoreKVPairs`.